mev_inspect/transfers.py

```diff
@@ -3,40 +3,57 @@
 from mev_inspect.classifiers.specs import get_classifier
 from mev_inspect.schemas.classifiers import TransferClassifier
 from mev_inspect.schemas.classified_traces import (
-    Classification,
     ClassifiedTrace,
     DecodedCallTrace,
 )
-from mev_inspect.schemas.transfers import ERC20Transfer, EthTransfer, TransferGeneric
+from mev_inspect.schemas.transfers import ETH_TOKEN_ADDRESS, Transfer
 from mev_inspect.traces import is_child_trace_address, get_child_traces
 
 
-def get_eth_transfers(traces: List[ClassifiedTrace]) -> List[EthTransfer]:
+def get_transfers(traces: List[ClassifiedTrace]) -> List[Transfer]:
     transfers = []
 
     for trace in traces:
-        if trace.value is not None and trace.value > 0:
-            transfers.append(EthTransfer.from_trace(trace))
+        transfer = get_transfer(trace)
+        if transfer is not None:
+            transfers.append(transfer)
 
     return transfers
 
 
-def get_transfers(traces: List[ClassifiedTrace]) -> List[ERC20Transfer]:
-    transfers = []
+def get_eth_transfers(traces: List[ClassifiedTrace]) -> List[Transfer]:
+    transfers = get_transfers(traces)
 
-    for trace in traces:
-        if isinstance(trace, DecodedCallTrace):
-            transfer = get_erc20_transfer(trace)
-            if transfer is not None:
-                transfers.append(transfer)
+    return [
+        transfer
+        for transfer in transfers
+        if transfer.token_address == ETH_TOKEN_ADDRESS
+    ]
 
-    return transfers
 
+def get_transfer(trace: ClassifiedTrace) -> Optional[Transfer]:
+    if trace.value is not None and trace.value > 0:
+        return _get_eth_transfer(trace)
```

Comment:
Note to self - this needs to change

This will represent any function call that's payable as a transfer. Should only be true transfers?

Reply:
I'm not sure you can distinguish between the two. e.g. an ETH -> stETH swap on Curve is a payable function.